Profiling features

reinago opened the pull request:
generalize profiling:
- enable logging
- preserve fast live info
- add code regions for modules
- nuke leftovers from previous approach
- bonus: interactive log console

reinago committed:
started adding PerformanceManager

reinago committed:
cannon size ++

reinago committed:
more thinking.

- extracting results is far too expensive at the moment

reinago committed:
should look better like this

reinago committed:
more problems, more solutions

geringsj commented:
As the profiling grows it may be time to merge the Frame Statistics Service into it. It seems they are doing similar things.

reinago committed:
multiple timer regions per frame

reinago committed:
missing include?

reinago committed:
too bad it does not work anymore

reinago committed:
like this. too bad it does not work anymore

reinago committed:
at least it compiles...

reinago committed:
guard against no started queries at all...

reinago committed:
more fiddling

reinago committed:
interface still a bit meh

reinago committed:
added profiling log config

reinago committed:
could produce a perf log shortly

reinago committed:
small fix

reinago committed:
I think I need some sleep

reinago committed:
add frameID

reinago committed:
first light of the profiling dump

reinago committed:
prepare parent pointer for fast lookup in GUI

reinago committed:
Merge branch 'master' into profiling

reinago committed:
resource grabbing idiom and IncludeAllGL fix

reinago committed:
Merge branch 'master' into profiling

reinago committed:
wow that was expensive

reinago committed:
formatting

reinago committed:
compilation fixes

reinago committed:
WIP passing the new perf data to GUI

reinago committed:
Merge branch 'master' into profiling

reinago committed:
Moved performance history keeping to GUI

- timers can be activated multiple times per frame
- draw a tube if min and max timings diverge
- removed old profiling stuff
- TODO: resurrect module timers when we can find them again.
- TODO: clean up unused profiling classes
- TODO: resurrect sliding-window statistics over the history

reinago committed:
this could possibly work...

reinago committed:
multi-frame statistics (that do not work yet?)

reinago committed:
fixed thinko and weird effect

reinago committed:
cleanup

reinago committed:
Fixed module map blowing up in GraphCollection

- added exemplary perf timer for Simple Sphere Renderer

reinago committed:
why not add the sum while we're at it

braunms committed:
Merge remote-tracking branch 'guido/profiling' into profiling

braunms committed:
GUI: moved graph syncro to graph collection

braunms committed:
fixed pointer handling for performance monitor, re-added lost part

braunms committed:
comment

reinago committed:
nuked old profiling

compiles with and without profiling now

braunms committed:
WIP added profiling gui window for modules

reinago committed:
Interactive console prototype

- still issues with focus
- large autocomplete is clunky

braunms committed:
Merge remote-tracking branch 'guido/profiling' into profiling

braunms committed:
WIP profiling gui for modules

braunms committed:
__really__ allow pop-ups of GUI windows only when GUIwindows are shown

braunms committed:
fixed profiling window zooming, rename  GUI defines

reinago committed:
bit of cleanup

reinago committed:
Modules work?

braunms committed:
usability of log console input, fixed module renaming bug leading to wrong parameter full names

braunms committed:
Merge remote-tracking branch 'reinago/profiling' into profiling

reinago committed:
commands fix

reinago committed:
formatting

reinago committed:
Merge branch 'master' into profiling

braunms committed:
minor stuff

reinago committed:
started fixing the merge mess

reinago committed:
more cleanup

braunms committed:
Merge remote-tracking branch 'reinago/profiling' into profiling

braunms committed:
merge fix and linux compile header fix

braunms committed:
restore minor merge hickup

braunms committed:
log console focus  stuff

braunms committed:
Log Console Input: fixed reclaiming of focus for closing pop-up, minor fix for mm command regex parsing

braunms committed:
minor rework of image button to allow toggle textures, added run-pause button for recording of profiling data

braunms committed:
Merge remote-tracking branch 'upstream/master' into profiling

braunms committed:
merge imagewidet changes to gl/non-gl changes

braunms committed:
non-gl mode compile fixes for profiling

braunms committed:
fix CI file check error

braunms committed:
clang formatting

braunms committed:
clang format cont.

braunms committed:
Merge remote-tracking branch 'upstream/master' into profiling

braunms committed:
pause update of profiling values globally if pause button is pressed in call or module

braunms committed:
implemented log console input history, added param hint for autocompleted

braunms committed:
improved parameter hint

braunms committed:
clang format

braunms committed:
log console ux fixes

braunms committed:
clang format

braunms committed:
cont.

reinago committed:
transient comment for timers

- fixed typo
- stringifier

braunms committed:
Merge remote-tracking branch 'upstream/master' into profiling

braunms committed:
wip profiling gui rework

reinago committed:
set the script path sooner (for self reflection and camera)

reinago committed:
moved lazy init of LogConsole to a safer place

tiny addition for more verbosity

Commit:
Format fix.

braunms committed:
Merge branch 'more_fixes' into profiling

braunms committed:
added new splitter for profiling data for the running graph in the configurator

braunms committed:
show profiling data of modules and calls in bottom tab bar in configurator

braunms committed:
minor profiling label formatting

braunms committed:
Merge remote-tracking branch 'upstream/master' into profiling

braunms committed:
Merge remote-tracking branch 'upstream/master' into profiling

braunms committed:
Merge remote-tracking branch 'upstream/master' into profiling

braunms committed:
Merge remote-tracking branch 'upstream/master' into profiling

braunms committed:
Merge remote-tracking branch 'upstream/master' into profiling

braunms committed:
merge and added drag and drop for profiling data

braunms committed:
added auto scroll while dragging profiling data

braunms committed:
added curve to connect profiling data with graph items

braunms committed:
style

braunms committed:
style

braunms committed:
style again

reinago committed:
compiles with profiling off
